docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +165,162 @@ Note that:
+
+To manage configuration files for child-devices the `c8y_configuration_plugin` supports two aspects:
+1) **Associating with cloud's child-device twin**<br/>
+Allowing to associate a configuration file with a cloud's child-device twin
+2) **Filetransfer from/to external device**<br/>
+Allowing to consume/provide a configuration file from/to an external device via network
+
+## Details to Aspect 1: Associating with cloud's child-device twin
+
+For aspect (1) the plugin provides the field `childid` for all records in the `c8y_configuration_plugin` configuration (reference to section 'Configuration' above). That field is interpreted as unique child-device id and the plugin associates the record's configuration file with corresponding cloud's child-device twin. If not provided the configuration file is associated with cloud's thin-edge device twin.
+
+Example:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },              # appears in the cloud on the thin-edge device
+    { path = '/etc/child1/foo.conf', type = 'foo', childid = 'child1'  }  # appears in the cloud on the child-device 'child1'
+  ]
+```
+
+## Details to Aspect 2: Filetransfer from/to external device
+
+For aspect (2) there is a proposal, followed by another alternative proposal. Decision has to been taken which proposal to follow.
+
+### Proposal: Filetransfer provided by tedge_agent, based on HTTP
+
+To provide/consume configuration files to/from external devices, the HTTP filetransfer feature of the `tedge_agent` is used.
+  
+The HTTP filetransfer feature of the `tedge_agent` provides the service to transfer files from external devices to the local filesystem of the thin-edge device's, and vice versa.
+
+  * `tedge_agent` serves PUT and GET requests on `http://<thin-edge IP address>/tedge/filetransfer/<path>`
+    * `path` is chosen by the external-device; it may contain folders (e.g. `foo/bar/config-file.conf`)
+    * on an incoming PUT request `tedge_agent` stores the contained file to `/http-root/<path>`
+    * on an incoming GET request `tedge_agent` sends the file `/http-root/<path>` to the requester (i.E. the external device)
+    * `tedge_agent` does not force any predefined folder structure within `/http-root/`. However for configuration management it is best practice to use folder structure as below:
+      * `/http-root/config/current/<child-id>/<config filename>`<br/>
+         for config-files provided by the external device to the thin-edge device (current state - or latest known state)<br/>
+      * `/http-root/config/desired/<child-id>/<config filename>`<br/>
+        for config-files to be consumed by the external device (desired state)
+  * `tedge_agent` notifies over MQTT any change of a file under `/http-root/` (e.g. when uploaded by a external-device or updated by a local process). The notification messages are published on the topic `tedge/filetransfer_change/{path}`.<br/>
```

Comment:
We are talking here about two different kinds of notifications:
1) notifications sent by tedge_agent's file-transfer feature, on every file change
When one (e.g. the external device) has changed/added a config-file within the folder that is under the control of the file-transfer feature (i.E. that http-root folder), tedge_agent will send a notification.
   - that is usually consumed by the c8y config plugin, to get recognized when the external device has uploaded a new version of a config-file to thin-edge

2) notification sent by c8y config plugin, when new config is available from cloud
    - that is usually consumed by the external device to get recognized when the plugin has a new config-file version delivered by c8y cloud

However you are right. In case 2 from above there are two notifications: That is since the tedge_agent will also send one, when the plugin places the new config-file into the http-root folder, to make it available for file-transfer to the external device.

Maybe it would be more consistent to let an external device stick on the file-transfer notifications (intead of one from the c8y config plugin), since the external device actually sticks also on file-transfer to download/upload the file from/to thin-edge device. That would mean the c8y config plugin could avoid to send any notification, when it puts the new config-file into the file-transfer's folder. With that we would also solve problem raised in https://github.com/thin-edge/thin-edge.io/pull/1236#discussion_r932075050.

@didier-wenzek , what do you think about that?

Reply:
The spec as of today has better MQTT topic structure for the point 1). So, I'd say, keep like that :D

Reply:
The point is, that the notification mechanism of the current plugin implementation is not enough for external child-devices. Since an external child-device does not share the same filesystem with the thin-edge device we need a kind of request-response pattern. That is e.g. in case of config retrieval operation from the cloud, to request the external child-device to upload it's config-file to the thin-edge device via HTTP file-transfer.

Moreover meanwhile we have an improved topic scheme for that request-response pattern.
See https://github.com/thin-edge/thin-edge.io/blob/313609b49a04863164c9d9045fa5f2e20c670c61/docs/src/references/c8y-configuration-management.md#fetching-configuration-file-from-child-device-to-cloud

and

https://github.com/thin-edge/thin-edge.io_spikes/blob/baf5c4f7082ae9cc64b9cd41e9366fac1ae3e2c6/file-transfer-service/SUMMARY.md#configuration-management-mqtt-apis

Please have a look.